DQMServices/Demo/test/TestDQMEDAnalyzer.cc

```diff
@@ -276,6 +276,58 @@ class TestDQMGlobalEDAnalyzer : public DQMGlobalEDAnalyzer<TestHistograms> {
 };
 DEFINE_FWK_MODULE(TestDQMGlobalEDAnalyzer);
 
+class TestDQMGlobalRunSummaryEDAnalyzer : public DQMGlobalRunSummaryEDAnalyzer<TestHistograms, int> {
+public:
+  explicit TestDQMGlobalRunSummaryEDAnalyzer(const edm::ParameterSet& iConfig)
+      : folder_(iConfig.getParameter<std::string>("folder")),
+        howmany_(iConfig.getParameter<int>("howmany")),
+        myvalue_(iConfig.getParameter<double>("value")) {}
+  ~TestDQMGlobalRunSummaryEDAnalyzer() override{};
+
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+    edm::ParameterSetDescription desc;
+    desc.add<std::string>("folder", "Global/testglobal")->setComment("Where to put all the histograms");
+    desc.add<int>("howmany", 1)->setComment("How many copies of each ME to put");
+    desc.add<double>("value", 1)->setComment("Which value to use on the third axis (first two are lumi and run)");
+    descriptions.add("testglobalrunsummary", desc);
+  }
+
+private:
+  std::shared_ptr<int> globalBeginRunSummary(edm::Run const&, edm::EventSetup const&) const {
+    return std::make_shared<int>(0);
+  }
+
+  void bookHistograms(DQMStore::IBooker& ibooker,
+                      edm::Run const&,
+                      edm::EventSetup const&,
+                      TestHistograms& h) const override {
+    h.folder = this->folder_;
+    h.howmany = this->howmany_;
+    h.bookall(ibooker);
+  }
+
+  void dqmAnalyze(edm::Event const& iEvent, edm::EventSetup const&, TestHistograms const& h) const override {
+    h.fillall(iEvent.luminosityBlock(), iEvent.run(), myvalue_);
+  }
+
+  void streamEndRunSummary(edm::StreamID, edm::Run const&, edm::EventSetup const&, int* runSummaryCache) const {
+    (*runSummaryCache) += 1;
+  }
+
+  void globalEndRunSummary(edm::Run const&, edm::EventSetup const&, int* runSummaryCache) const {}
+
+  void dqmEndRun(edm::Run& run,
+                 edm::EventSetup const& setup,
+                 TestHistograms const& h,
+                 int const& runSummaryCache) const {}
```

Comment:
Note that there is some inevitable interface inconsistency: I chose to pass the `RunSummaryCache` object here by reference, while the `streamEndRunSummary` and `globalEndRunSummary` take it by pointer (directly from the framework).

Another question is if users should be required to override `globalEndRunSummary()` (as in here), or should the `DQMGlobalRunSummaryEDAnalyzer` base class provide an empty default implementation (as it does for `dqmEndRun()`.